messaging/src/main/java/org/axonframework/eventhandling/deadletter/DeadLetteringEventHandlerInvoker.java

```diff
@@ -0,0 +1,280 @@
+        private boolean allowReset = false;
+        private ListenerInvocationErrorHandler listenerInvocationErrorHandler = PropagatingErrorHandler.instance();
+
+        private Builder() {
+            // The parent's error handler defaults to propagating the error.
+            // Otherwise, faulty events would not be dead-lettered.
+            super.listenerInvocationErrorHandler(PropagatingErrorHandler.instance());
+        }
+
+        /**
+         * Sets the {@link DeadLetterQueue} this {@link EventHandlerInvoker} maintains dead-letters with.
+         *
+         * @param queue The {@link DeadLetterQueue} this {@link EventHandlerInvoker} maintains dead-letters with.
+         * @return The current Builder instance for fluent interfacing.
+         */
+        public Builder queue(@Nonnull DeadLetterQueue<EventMessage<?>> queue) {
+            assertNonNull(queue, "The DeadLetterQueue may not be null");
+            this.queue = queue;
+            return this;
+        }
+
+        /**
+         * Sets the processing group name of this invoker.
+         *
+         * @param processingGroup The processing group name of this {@link EventHandlerInvoker}.
+         * @return The current Builder instance for fluent interfacing.
+         */
+        public Builder processingGroup(@Nonnull String processingGroup) {
+            assertNonEmpty(processingGroup, "The processing group may not be null or empty");
+            this.processingGroup = processingGroup;
+            return this;
+        }
+
+        /**
+         * Sets the {@link TransactionManager} used by this invoker for <b>all</b> operations involving the configured
+         * {@link DeadLetterQueue}.
+         *
+         * @param transactionManager The {@link TransactionManager} used by this invoker for <b>all</b> operations
+         *                           involving the configured {@link DeadLetterQueue}
+         * @return The current Builder instance for fluent interfacing.
+         */
+        public Builder transactionManager(@Nonnull TransactionManager transactionManager) {
+            assertNonNull(transactionManager, "The TransactionManager may not be null");
+            this.transactionManager = transactionManager;
+            return this;
+        }
+
+        /**
+         * Sets whether this {@link DeadLetteringEventHandlerInvoker} supports resets of the provided
+         * {@link DeadLetterQueue}. If set to {@code true}, {@link DeadLetterQueue#clear(String)} will be invoked upon a
+         * {@link #performReset()}/{@link #performReset(Object)} invocation. Defaults to {@code false}.
+         *
+         * @param allowReset A toggle dictating whether this {@link DeadLetteringEventHandlerInvoker} supports resets of
+         *                   the provided {@link DeadLetterQueue}.
+         * @return The current Builder instance for fluent interfacing.
+         */
+        public Builder allowReset(boolean allowReset) {
```

Comment:
There is currently no easy means to configure this property, as the `EventProcessingConfigurer` constructs this class.
Perhaps a similar configuration option as with the `PooledStreamingEventProcessor`, a `Function` that provides the builder and reurns the builder as additional configuration makes sense.
Thoughts?

Reply:
This is done, through means of allowing the proposed configuration options.